### PR TITLE
Fix keyboard shortcuts interferring with input fields [BC]

### LINKED ISSUE #371: Unable to type "-" or "=" in input fields outside BR
To reproduce:
1. Go to https://archive.org/details/completebookofpi00hass
2. Try to type "-" or "=" into the search box at the top of the page

* Expected: Can type them
* Actual: They zoom in/out the BR

## Proposal & Constraints

- Keyboard shortcuts should only work if the event target is not an input box

## Stakeholders

@textfiles

### PR DESCRIPTION
**Pull request by cdrini**
Closes #371 ; Closes #673 ; Closes WEBDEV-4107

Breaking change: Remove public method `BookReader.prototype.keyboardNavigationIsDisabled`

To test:

Go to one of the demos.

Run this JS:

```js
customElements.define('test-element', class extends HTMLElement {
  constructor() {
    super();
    this.attachShadow({mode: 'open'});
    this.shadowRoot.append(document.createElement('input'));
  }
});
$(document.body).prepend($(`<test-element/>`));
```

Type in the added input.

- [x] Confirm keyboard shortcuts work when the input does not have focus
- [x] Confirm keyboard shortcuts do not work when the input does have focus
- [x] Confirm keyboard shortcuts do not work when typing in searchbox in side panel
- [x] Confirm keyboard shortcuts do not work when typing in searchbox in header

### CODE REVIEW

File: src/BookReader/utils.js [hunk 1]
```diff
@@ -40,6 +40,23 @@ export function notInArray(value, array) {
...
 }
 
+/**
+ * Determines the active element, going into shadow doms.
+ * @return {Element}
+ */
+export function getActiveElement(doc = document, recurseShadowDom = true) {
+  const activeElement = doc.activeElement;
+  if (recurseShadowDom && activeElement?.shadowRoot) {
+    return getActiveElement(activeElement.shadowRoot, true);
```
Comment (iisa) [inline src/BookReader/utils.js:50]: smol q: this shouldn't be too expensive, ya?

**Review by iisa — APPROVED** [review]
looks good! thanks for jumping on this.
had 1 q

File: src/BookReader/utils.js (hunk 1 above)
Comment (cdrini) [reply src/BookReader/utils.js:50]: Hopefully? :P It should only have to go through I'd guess like max 10? We might want to optimize to only run this if the key pressed is on of interest. Right now it runs for any kepress, before determining if it's of interest. Let me see how hard that would be...

Comment (iisa) [reply src/BookReader/utils.js:50]: maybe if we debounce/throttle that keypress event handler, it _may_ be enough..

Comment (jbuckner) [reply src/BookReader/utils.js:50]: Is there a way to be more explicit about this than traversing the DOM? ie add a method to the components that can be active and call `component.isActive()`?

Comment (cdrini) [reply src/BookReader/utils.js:50]: @iisa Switched order of calls :+1:

We'd need something like `isKeyboardSink()` or something. This isn't the _best_ solution as is; I think there are ways one can fully block shadow DOM access to your web component. But it solves the current problem for our components, and we can always change it later :) This also has the benefit of working for other people who might embed BR in their sites, which might not be using our web component methods.

**Pull request merged**